### PR TITLE
Implement support for specifying node protocol in CLI (V2)

### PR DESCRIPTION
**Pull request by dcoutts**
This means that on the command line you can select if you're talking to a node
that is in Byron-only mode, Shelley-mode or the composite Cardano mode.

We have to handle the queries slightly differently in these cases.

### CODE REVIEW

**Comment by dcoutts** [comment]
So far this only converts one command, but it should show the pattern to follow. I'll do tx submission next.

This incorporates many of the changes from #1427 but based on updated APIs.

File: cardano-node/cardano-node.cabal [hunk 1]
```diff
@@ -202,6 +202,7 @@ executable chairman
                      , containers
                      , cardano-api
                      , cardano-config
+                     , cardano-ledger
```
Comment (Jimbo4350) [inline cardano-node/cardano-node.cabal:205]: @dcoutts shouldn't this be the ledger specs repo?

Comment (dcoutts) [reply cardano-node/cardano-node.cabal:205]: It's the `cardano-ledger` package that lives in the ledger specs repo in `byron/ledger/impl/`

Comment (mrBliss) [reply cardano-node/cardano-node.cabal:205]: Indeed

File: cardano-api/src/Cardano/Api/LocalChainSync.hs [hunk 2]
```diff
@@ -17,15 +17,15 @@ import           Ouroboros.Network.Protocol.ChainSync.Client
...
-        localChainSyncClient = Just (chainSyncGetCurrentTip resultVar)
-      }
-    atomically (takeTMVar resultVar)
+    resultVar <- liftIO newEmptyTMVarIO
+    liftIO $ connectToLocalNode
+               connctInfo
+               nullLocalNodeClientProtocols {
+                 localChainSyncClient = Just (chainSyncGetCurrentTip resultVar)
+               }
+    liftIO $ atomically (takeTMVar resultVar)
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/LocalChainSync.hs:28]: Is this change necessary?

Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/LocalChainSync.hs:28]: Not if you can figure out how to satisfy GHC :slightly_smiling_face:
